File: server/router/business/sysApis.go
package business

import (
	v1 "go-cms/api/v1"
	"go-cms/middleware"

	"github.com/gin-gonic/gin"
)

type SysApisRouter struct {
}

// InitSysApisRouter 初始化 SysApis 路由信息
func (s *SysApisRouter) InitSysApisRouter(Router *gin.RouterGroup) {
	router := Router.Group("sysApis").Use(middleware.OperationRecord())
	routerNoRecord := Router.Group("sysApis")
	var apiV1 = v1.ApiGroupApp.BusinessApiGroup.SysApisApi
	{
		router.POST("createSysApis", apiV1.CreateSysApis)           // 新建SysApis
		router.POST("deleteSysApis", apiV1.DeleteSysApis)           // 删除SysApis
		router.POST("deleteSysApisByIds", apiV1.DeleteSysApisByIds) // 批量删除SysApis
		router.POST("updateSysApis", apiV1.UpdateSysApis)           // 更新SysApis
		router.POST("quickEdit", apiV1.QuickEdit)                   // 快速编辑
	}
	{
		routerNoRecord.GET("findSysApis", apiV1.FindSysApis)       // 根据ID获取SysApis
		routerNoRecord.GET("getSysApisList", apiV1.GetSysApisList) // 获取SysApis列表
		routerNoRecord.GET("excelList", apiV1.ExcelList)           // 分页导出excel SysApis列表
	}
}

File: web/src/api/sysApis.ts
import { http,resData } from "@/utils/http";



// @Tags SysApis
// @Summary 创建SysApis
export const createSysApis = (data?: object) => {
  return http.request<resData>("post", "/sysApis/createSysApis", { data });
};

// @Tags SysApis
// @Summary 删除SysApis
export const deleteSysApis = (data?: object) => {
  return http.request<resData>("post", "/sysApis/deleteSysApis", { data });
};


// @Tags SysApis
// @Summary 删除SysApis
export const deleteSysApisByIds = (data?: object) => {
  return http.request<resData>("post", "/sysApis/deleteSysApisByIds", { data });
};


// @Tags SysApis
// @Summary 更新SysApis
export const updateSysApis = (data?: object) => {
  return http.request<resData>("post", "/sysApis/updateSysApis", { data });
};


// @Tags SysApis
// @Summary 用id查询SysApis
export const findSysApis =  (params?: object) => {
  return http.request<resData>("get", "/sysApis/findSysApis", { params });
};


// @Tags SysApis
// @Summary 分页获取SysApis列表
export const getSysApisList =  (params?: object) => {
  return http.request<resData>("get", "/sysApis/getSysApisList", { params });
};




// @Tags SysApis
// @Summary 快速编辑
export const quickEdit = (data?: object) => {
  return http.request<resData>("post", "/sysApis/quickEdit", { data });
};


// @Tags SysApis
// @Summary 导出excelSysApis列表
export const excelList =(data?: object) => {
  return http.request<resData>("get", "/sysApis/excelList", { data });
};




